### ISSUE TITLE
feature extract using CNN  with unconstrained  length image

### ISSUE THREAD

**Issue by ghhong1986** [weinman/cnn_lstm_ctc_ocr#7]
hi weinman , I have read the paper and code and try to understand  but a few questions confused me, please help me.
1.  the model we input data by function  <code>bucket_by_sequence_length</code> with paramter dynamic_pad  setted True , in every batch has a fix shape , but different batch may have different shape, so how does cnn in the model  work ?  
2.  how to write  inference service  when input  different width images ?
3.  Any theroy about sequence length calculation in   end of convnet layer?
thanks.

**Comment by weinman** [comment]
1.  The cnn is applied to the entire image, including the padding. The CTC layer indicates the sequence length; as a result I am not entirely certain whether the gradient for the padded region is included in the loss function. Even if it is, it should not contribute much because it should not change the loss (due to the sequence length restriction).

2.  I am not sure what you are asking. The CTC layers ask for the sequence length precisely so it does not utilize irrelevant logits that arise due to padding.

3. This calculation captures how the downsizing and padding changes the length of the original, unpadded image width so that you know which timesteps are valid in the final 1D sequence. I make the calculation step-by-step so it is easier to identify (and verify) what the image width is after each layer's transformation.
